### ISSUE TITLE
edata inconsists with dstdata and srcdata in block

### ISSUE THREAD

**Issue by patricxu** [dmlc/dgl#2540]
The node of edata is not in the set of dstdata and srcdata when use block.

![image](https://user-images.githubusercontent.com/9252137/104586455-73a6cf00-56a0-11eb-89aa-97d28357daab.png)

As you can see from the picture, the edge should be ([104, 105], [105, 104]). but i got ([1,0], [0,1]) instead.

Thanks!

**Comment by patricxu** [comment]
I found a bug in your demo code which caused the issue.
[https://docs.dgl.ai/guide/minibatch-custom-sampler.html#implementing-a-custom-neighbor-sampler](url)

The MultiLayerDropoutSampler should be like this.
`class MultiLayerRondomWalkSampler(dgl.dataloading.BlockSampler):
    def __init__(self, n_layers, restart_prob=0.8, return_eids=False):
        super().__init__(n_layers, return_eids)
        self.restart_prob = restart_prob

    def sample_frontier(self, block_id, g, seed_nodes):
        src, dst = dgl.in_subgraph(g, seed_nodes).all_edges()
        if block_id == self.num_layers - 1:
            restart_prob = 0
        else:
            restart_prob = self.restart_prob
        mask = torch.zeros_like(src).bernoulli_(1 - restart_prob)
        src = src[mask==1]
        dst = dst[mask==1]
        num_nodes = torch.unique(torch.cat((src, dst))).size()
        frontier = dgl.graph((src, dst), num_nodes=g.number_of_nodes())
        return frontier`

Note that src = src[mask == 1], dst = dst[mask==1] rather than src = src[mask], dst = dst[mask]
